c2rust-analyze/src/c_void_casts.rs

```diff
@@ -274,16 +287,88 @@ impl<'tcx> CVoidCasts<'tcx> {
     /// [`Cast`]: rustc_middle::mir::Rvalue::Cast
     /// [`From`]: CVoidCastDirection::From
     /// [`To`]: CVoidCastDirection::To
-    pub fn should_skip_stmt(&self, stmt: &Statement<'tcx>) -> bool {
-        let (lhs, rv) = match get_assign_sides(stmt) {
-            None => return false,
-            Some(it) => it,
-        };
-        self.to.contains(lhs)
-            || self.from.contains(match get_cast_place(rv) {
-                None => return false,
-                Some(it) => it,
-            })
+    pub fn should_skip_stmt(&self, loc: &Location) -> bool {
+        self.to.contains(loc) || self.from.contains(loc)
```

Comment:
Can you please add a separate field to `CVoidCastsUniDirectional` for tracking "casts to skip", separate from the map for tracking "calls with a cast on the argument/result"?  Right now, an entry in the `CVoidCastsUniDirectional` map means "either this is a call, and has the following cast on its argument/result, OR this is a cast, and it should be skipped".  This will work, but IMO it would be clearer to have separate maps/sets for the two cases.

Reply:
Is this why there are two insertions into `c_void_casts`?  I agree it would be much clearer to separate the two.

Reply:
d2cd89bf